server/planning/events/events_spike.py

```diff
@@ -1,320 +1,335 @@
+    else:
+        spiked_events = past + future
 
+    notifications = []
+    for event in spiked_events:
+        if not can_spike_event(event, events_with_plans):
+            continue
+
+        new_updates = {"skip_on_update": True}
+        spike_event(new_updates, event)
+        await events_service.update(event[ID_FIELD], new_updates)
+        item = await events_service.find_by_id_raw(event[ID_FIELD])
+        await signals.event_spiked.send(new_updates, event)
+
+        if item:
             notifications.append(
                 {
                     "id": event[ID_FIELD],
                     "etag": item["_etag"],
-                    "state": event.get("revert_state", WORKFLOW_STATE.DRAFT),
+                    "revert_state": item["revert_state"],
                 }
             )
 
-        updates["_unspiked_items"] = notifications
+    updates["_spiked_items"] = notifications
 
-    def update(self, id, updates, original):
-        unspiked_items = updates.pop("_unspiked_items", [])
-        item = super().update(id, updates, original)
 
-        if self.is_original_event(original):
-            user = get_user(required=True).get(ID_FIELD, "")
-            unspiked_items.append({"id": id, "etag": item["_etag"], "state": item[ITEM_STATE]})
+async def unspike_recurring_events(updates: dict[str, Any], original: dict[str, Any], update_method: str) -> None:
+    events_service = EventResourceModel.get_service()
 
-            push_notification(
-                "events:unspiked",
-                item=str(original[ID_FIELD]),
-                user=str(user),
-                unspiked_items=unspiked_items,
-            )
+    historic, past, future = await get_recurring_timeline(original, spiked=True)
+    remove_lock_information(updates)
+    unspike_event(updates, original)
+
+    # Determine if the selected event is the first one, if so then
+    # act as if we're changing future events
+    if len(historic) == 0 and len(past) == 0:
+        update_method = UPDATE_FUTURE
+
+    if update_method == UPDATE_FUTURE:
+        unspiked_events = future
+    else:
+        unspiked_events = past + future
+
+    notifications = []
+    for event in unspiked_events:
+        if event.get(ITEM_STATE) != WORKFLOW_STATE.SPIKED:
+            continue
+
+        new_updates = {"skip_on_update": True}
+        unspike_event(new_updates, event)
+        await events_service.update(event[ID_FIELD], new_updates)
+        item = await events_service.find_by_id_raw(event[ID_FIELD])
+        await signals.event_unspiked.send(new_updates, event)
+
+        notifications.append(
+            {
+                "id": event[ID_FIELD],
+                "etag": item["_etag"],
+                "state": event.get("revert_state", WORKFLOW_STATE.DRAFT),
+            }
+        )
+
+    updates["_unspiked_items"] = notifications
+
+
+async def process_spike_event(updates: dict[str, Any], original: dict[str, Any]) -> dict[str, Any] | None:
```

Comment:
We can safely assume this function will return a `dict`, and never `None`, this is because your're returning the result of a find_by_id and we have already done checks to make sure the Event exists.

Therefor the return type can simply be a dict, and you can maybe place an assert after `spiked_event = await events_service.find_by_id_raw(original[ID_FIELD])`